Mathlib/AlgebraicGeometry/EllipticCurve/NormalForms.lean

```diff
@@ -0,0 +1,667 @@
+/-
+Copyright (c) 2024 Jz Pan. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jz Pan
+-/
+import Mathlib.AlgebraicGeometry.EllipticCurve.VariableChange
+import Mathlib.Algebra.CharP.Defs
+
+/-!
+
+# Some normal forms of elliptic curves
+
+This file defines some normal forms of Weierstrass equations of elliptic curves.
+
+## Main definitions and results
+
+The following normal forms are in [silverman2009], section III.1, page 42.
+
+- `WeierstrassCurve.IsCharNeTwoNF` is a predicate which asserts that a `WeierstrassCurve` is
+  of form $Y^2 = X^3 + a_2X^2 + a_4X + a_6$. It is the normal form of characteristic ≠ 2.
+
+  If 2 is invertible in the ring (for example, if it is a field of characteristic ≠ 2),
+  then for any `WeierstrassCurve` there exists a change of variables which will change
+  it into such normal form (`WeierstrassCurve.exists_variableChange_isCharNeTwoNF`).
+  See also `WeierstrassCurve.toCharNeTwoNF` and
+  `WeierstrassCurve.toCharNeTwoNF_spec`.
+
+The following normal forms are in [silverman2009], Appendix A, Proposition 1.1.
+
+- `WeierstrassCurve.IsCharNeTwoThreeNF` is a predicate which asserts that a `WeierstrassCurve` is
+  of form $Y^2 = X^3 + a_4X + a_6$. It is the normal form of characteristic ≠ 2.
```

Comment:
```suggestion
  of form $Y^2 = X^3 + a_4X + a_6$. It is the normal form of characteristic ≠ 2 or 3.
```

If you decide to call this `WeierstrassCurve.IsCharThreeJZeroNF` then you should rearrange the sentences.